static_src/test/unit/util/poll.spec.js

```diff
@@ -0,0 +1,46 @@
+
+import '../../global_setup.js';
+
+import poll from '../../../util/poll';
+
+describe('poll()', function() {
+  const condition = function condition(x) { return !!x; };
+  let promise;
+  let request;
+
+  beforeEach(function() {
+    request = function() {
+      return new Promise(function(resolve, reject) {
+        promise = {resolve, reject};
+      });
+    }
+  });
+
+  it('should resolve returned promise when condition is true', function(done) {
+    let value;
+    const actual = poll(condition, request, 0, 1);
+    actual.then((res) => value = res);
+    promise.resolve(true);
+
+    setImmediate(function() {
+      expect(value).toEqual(true);
+      done();
+    });
+  });
+
+  it('should re-send the promise if the condition is false', function() {
+    // TODO not sure how to test
+  });
```

Comment:
Yeah, this is pretty tricky, you could do something do something like you have above and use [sinon's fake timers](http://sinonjs.org/docs/#clock) to tick the clock forward to trigger the poll. Here's some pseudo code:

```javascript
let lastCapturedPromise;
let clock = sinon.useFakeTimers();
function capturedPromiseFactory () { // return a promise, save the resolve/reject to lastCapturedPromise }

it(funciton() {
  pollPromise = poll(identity, capturedPromiseFactory);
  lastCapturedPromise.resolve(false);
  clock.tick(1);
  // poll runs again, lastCapturedPromise is updated
  // assert pollPromise not resolved
  lastCapturedPromise.resolve(true)
  clock.tick(1)
  // assert pollPromise is resolved
});
```

I think the promises resolve async, so that adds more complexity because you have to do your asserts async as well :(